lib/Sema/DerivedConformanceDistributedActor.cpp

```diff
@@ -156,6 +154,36 @@ static ValueDecl *deriveDistributedActor_actorTransport(
   return propDecl;
 }
 
+static Type deriveDistributedActor_Transport(
+    DerivedConformance &derived) {
+  assert(derived.Nominal->isDistributedActor());
+  auto &C = derived.Context;
+
+  // Look for a type DefaultActorTransport within the parent context.
+  auto defaultTransportLookup = TypeChecker::lookupUnqualified(
+      derived.getConformanceContext()->getModuleScopeContext(),
+      DeclNameRef(C.Id_DefaultActorTransport),
+      derived.ConformanceDecl->getLoc());
+  TypeDecl *defaultTransportTypeDecl = nullptr;
+  for (const auto &found : defaultTransportLookup) {
+    if (auto foundType = dyn_cast_or_null<TypeDecl>(found.getValueDecl())) {
+      if (defaultTransportTypeDecl) {
+        // Note: ambiguity, for now just fail.
+        return nullptr;
+      }
+
+      defaultTransportTypeDecl = foundType;
+      continue;
+    }
+  }
+
+  // There is no default, so fail to synthesize.
+  if (!defaultTransportTypeDecl)
+    return nullptr;
+
+  // Return the default transport type.
+  return defaultTransportTypeDecl->getDeclaredInterfaceType();
+}
```

Comment:
So... this means that we look in the current module for `DefaultActorTransport`, and that dictates the type of the `Transport` property... I remain worried about this somehow... What happens if there is no default one defined, you'd suggest that each actor must explicitly define the typealias?

I was hoping we can derive it from the init declarations:

```swift

distributed actor A {
  init(t: MyTransport) {}
  // therefore:
  typealias Transport = MyTransport
}
```

and also once all transport types must "match":

```swift
distributed actor A {
  init(t: MyTransport) {}
  init(number: Int, other: OtherTransport) {}
  // ERROR: distributed actor initializers must accept 
  // the same transport type in all initializers
```

The `DefaultActorTransport` we could still keep though! It's really be the default, but this init declaration dance I wonder if would not be worth it...? Maybe let's ignore for now hm...

Reply:
As I've noted in my newly-written comment, I expect that any library providing a transport would provide a `DefaultActorTransport` typealias pointing at that transport, so importing the library + defining a distributed actor would use that transport. Inference from `init(transport:...)` would also be possible.